helm/prometheus-operator-app/values.yaml

```diff
@@ -1350,7 +1350,7 @@ prometheusOperator:
 
   ## Set the prometheus config reloader side-car CPU limit
   ##
-  configReloaderCpu: 100m
+  configReloaderCpu: 25m
```

Comment:
We're syncing this chart with upstream. Can we justify it and propose change upstream https://github.com/helm/charts/blob/master/stable/prometheus-operator/values.yaml#L1346 ?

Reply:
Justification in linked issue seems sensible to me. Given the targeted service I can't see how the resource usage would change too much, even under load.
Generally I'd prefer to set these values per instance for now and to request this upstream. Otherwise syncing with upstream gets more tedious since with the current process this change will also show up on every future commit list.

Reply:
For now we observed that both `prometheus-config-reloader` and `rules-configmap-reloader` container have very low cpu usage, below 0.0001.

![image](https://user-images.githubusercontent.com/6536819/87137819-cccf1980-c29d-11ea-95a7-4dfda8e45517.png)

We just want to get going with the new monitoring stack, lowering cpu here help to go from 300m cpu to 150m cpu per prometheus so it can fit on every CP with minimal impact. We might adjust this in the future.

Reply:
Let's collect more data and real cases, find a value we are comfortable with and then we might propose a change upstream.

Reply:
Would it make sense to:
- configure configuration override where needed only, using user values ConfigMap support, and
- if generally applicable create a PR for upstream?

Reply:
> Would it make sense to:
> 
> * configure configuration override where needed only, using user values ConfigMap support, and
> * if generally applicable create a PR for upstream?

I agree with both these points. It wouldn't be that much more to just update a user values file. This would put less pressure on Halo to maintain this app.... That's what user values are for. 

Getting it into upstream would be nice, but I am not sure e have a good argument for them to merge it.

So let's just put it in user values file? @TheaMarie Is there any reason why we can't do that?

Reply:
There's no good argument for now to propose this change upstream. I suggest we go with this and later on when we have collected more data from real world usage we can decide to propose a change upstream.

I don't get this "user values file", I've updated the `values.yaml` here that's where override for helm chart takes place usually. I'm lost could you please explain what you mean by user value here and ConfigMap support ?

Reply:
values.yaml in chart has defaults. Those defaults can be overridden on multiple levels, including but not limited to:
- via app catalog ConfigMap (which can be just like the catalog - global, or installation specific override)
- via user values ConfigMap (so has to be done for each helm release where needed)

See https://intranet.giantswarm.io/docs/dev-and-releng/app-developer-processes/managing_app_configuration/ for more details.

Proposal is to keep values.yaml in chart unmodified when possible, so it's easy to sync with upstream chart. Instead this particular customization could be done in app catalog CM (if it's supposed to be applied to all CPs) or in user values CM (if it's supposed to be tuned just for specific helm release)

E.g. recently Nick made kiam app region configurable in helm chart
https://github.com/giantswarm/kiam-app/pull/22/files#diff-a3c904e565c362cb2c2d54495074f49dR99-R101
and then customized it for all aws installations via installation specific app catalog CMs - https://github.com/giantswarm/installations/pull/1326/files

Similar can be done for every App CR - define user values CM, use it in App CR and through this override anything from the chart for that helm release only.

Reply:
Btw global (that apply to all installations) appcatalog defaults are configured in https://github.com/giantswarm/installations/tree/master/default/appcatalog

Reply:
That is really cool, I didn't know about those overrides. Very nice.

Reply:
Kudos to @rossf7 and @tomahawk28 of batman fame for the app infra and config options!